book/chrome.md

```diff
@@ -119,267 +130,178 @@ Here\'s how I want inline layout to work, at a high level:
     position and call `layout` on each `TextLayout`
 -   `TextLayout.layout` will compute an `x` and a `y` position as well
 
-To begin with, we\'ll need to create two new data structures:
+To begin with, we'll need to create two new data structures:
 
 ``` {.python}
 class LineLayout:
-    def __init__(self, parent):
+    def __init__(self, node, parent):
+        self.node = node
         self.parent = parent
         self.children = []
-        self.w = 0
+        self.cx = 0
 
 class TextLayout:
-    def __init__(self, node, text):
-        self.children = []
+    def __init__(self, node, word):
         self.node = node
-        self.text = text
+        self.children = []
+        self.word = word
 ```
 
-`LineLayout` is pretty run-of-the-mill here,[^3] but `TextLayout` is
-unusual. First, it\'s got a dummy `children` field. I added that just to
-keep it uniform; it\'ll always be an empty list.[^4] But then, it\'s
-also got both a `node` and a `text` argument. That\'s because a single
-`TextNode` contains multiple words, and I want each `TextLayout` to be a
-single word.[^5] Finally, I\'m not attaching the `TextLayout` to its
-parents. I\'ll do that in a separate `attach` method, which I\'ll define
-below.
-
-[^3]: There\'s no dummy `node` field on `LineLayout` because there\'s no
-    HTML node that corresponds to a line of text, and there couldn\'t
-    be.
+`LineLayout` is pretty run-of-the-mill, but `TextLayout` is unusual.
+First, its `children` field will always be an empty list; that's just
+for convenience.[^4] Second, it has both a `node` and a `word`
+argument. That's because a single `TextNode` contains multiple words,
+and I want each `TextLayout` to be a single word.[^5] Finally,
+`TextLayout` does not take a `parent` argument. That's because we'll
+decide the parent later, in a separate `append` method, which I'll
+define below.
 
 [^4]: You may want to use inheritance to group all the `Layout`
-    classes into a hierarchy, but I\'m trying to stick to some kind of
+    classes into a hierarchy, but I'm trying to stick to some kind of
     easily-translatable subset of Python.
 
-[^5]: Because of line breaking. Note that in real browsers, even words can
-have line breaks in them, via hyphenation.
```

Comment:
Why remove this?

Reply:
It wasn't removed, just moved elsewhere in the document. My initial Markdown-ification of the book was via Pandoc, which puts footnotes at the bottom. I've tried to move them near where they're used instead.